mathml/presentation-markup/mrow/mrow-fallback-002.html

```diff
@@ -0,0 +1,151 @@
+      case "munderover":
+      case "msubsup":
+        return mspaceCount == 3;
+      case "mmultiscripts":
+        return mspaceCount % 2 == 1;
+    }
+  }
+
+  function wrapShrink(el) {
+    let wrapper = document.createElement("div");
+    let math = document.createElement("math");
+    wrapper.style.display = "inline-block";
+    math.appendChild(el);
+    wrapper.appendChild(math);
+    return wrapper;
+  }
+
+  function runTests() {
+    let container = document.getElementById("container");
+    const epsilon = 1;
+    ["mfrac", "mroot", "munder", "mover", "munderover", "msub", "msup", "msubsup", "mmultiscripts"].forEach(tag => {
+      let element = FragmentHelper.createElement(tag);
+      let reference = FragmentHelper.createElement("mrow");
+      let elementWrapper = wrapShrink(element);
+      let referenceWrapper = wrapShrink(reference);
+      container.appendChild(elementWrapper);
+      container.appendChild(referenceWrapper);
+      let maxCount = tag == "mmultiscripts" ? 10 : 5;
+      let mspaceCount = 0;
+      for (let count = 0; count <= maxCount; count++) {
+        if (!isValid(tag, mspaceCount)) {
+          test(function() {
+            assert_approx_equals(elementWrapper.getBoundingClientRect().width, referenceWrapper.getBoundingClientRect().width, epsilon);
+          }, `Invalid <${element.tagName}> should fallback to mrow behaviour when computing preferred logical widths (count == ${count})`);
+        }
+        if (tag == "mmultiscripts" && count == maxCount / 2) {
+          [element, reference].forEach(el => {
+            el.insertAdjacentHTML("beforeend", `<mprescripts/>`);
+          });
+        } else {
+          let width = (count + 1) * 10;
+          let height = (count + 1) * (count % 2 ? 15 : 5);
+          let depth = (count + 1) * (count % 2 ? 5 : 15);
+          [element, reference].forEach(el => {
+            el.insertAdjacentHTML("beforeend", `<mspace height="${height}px" depth="${depth}px" width="${width}px" style="background: black"/>`);
+          });
+          mspaceCount++;
+        }
+      }
+    });
+    Array.from(document.getElementById("invalidMultiscripts").
+               getElementsByClassName("invalid-mmultiscripts-container")).forEach(element => {
+      let reference = element.nextElementSibling;
+      let description = element.dataset.description;
+      console.log(reference, description)
+      test(function() {
+        assert_approx_equals(element.getBoundingClientRect().width, reference.getBoundingClientRect().width, epsilon);
+      }, `Invalid mmultiscripts should fallback to mrow behaviour when computing preferred logical widths (${description})`);
+    });
+
+    done();
+  }
+</script>
+</head>
+<body>
+  <div id="log"></div>
+  <div id="container"></div>
+  <div id="invalidMultiscripts">
+    <div style="background: lightgreen; display: inline-block" class="invalid-mmultiscripts-container" data-description="first in-flow child is an <mprescripts>">

```

Comment:
I believe background: lightgreen is not necessary but maybe that's useful for developers debugging the tests. Up to you but please make it consistent with `wrapShrink()`. Also better to use a more neutral color such as gray as green often means PASS in WPT.

Reply:
Removed in `invalid-markup.js`